src/Cli/dotnet/commands/dotnet-test/Terminal/TerminalTestReporter.cs

```diff
@@ -1024,7 +1053,7 @@ public void TestInProgress(
         string displayName,
         string? executionId)
     {
-        TestProgressState asm = _assemblies[$"{assembly}|{targetFramework}|{architecture}|{executionId}"];
+        TestProgressState asm = (TestProgressState)_assemblies[$"assembly|{assembly}|{targetFramework}|{architecture}|{executionId}"];
```

Comment:
Is this expected text? `assembly|Contoso.Test.dll|net10.0|x64|XX`?

Reply:
yes, that is the key